landlab/layers/eventlayers.py

```diff
@@ -661,16 +869,147 @@ def add(self, dz, **kwds):
             try:
                 self[name][-1] = kwds[name]
             except KeyError:
-                msg = "MaterialLayers: {0} is not being tracked. Error in adding.".format(
-                    name
+                raise ValueError(
+                    "EventLayers: {0} is not being tracked. Error in adding.".format(
+                        name
+                    )
                 )
-                raise ValueError(msg)
+
+    def reduce(self, *args, **kwds):
+        """reduce([start], stop, [step])
+        Combine layers.
+
+        Reduce adjacent layers into a single layer.
+
+        Examples
+        --------
+        >>> from landlab.layers.eventlayers import EventLayers
+
+        Create an empty layer stack with 3 stacks.
+
+        >>> layers = EventLayers(3)
+        >>> layers.number_of_layers
+        0
+
+        To add a layer of uniform thickness to every stack.
+
+        >>> layers.add(1.5)
+        >>> layers.number_of_layers
+        1
+        >>> layers.dz
+        array([[ 1.5,  1.5,  1.5]])
+
+        Add a second layer with uneven thickness.
+
+        >>> layers.add([1., 2., .5])
+        >>> layers.dz
+        array([[ 1.5,  1.5,  1.5],
+               [ 1. ,  2. ,  0.5]])
+
+        >>> layers.reduce()
+        >>> layers.dz
+        array([[ 2.5,  3.5,  2. ]])
+
+        >>> layers.add([1., 2., .5])
+        >>> layers.add([1., 2., .5])
+        >>> layers.dz
+        array([[ 2.5,  3.5,  2. ],
+               [ 1. ,  2. ,  0.5],
+               [ 1. ,  2. ,  0.5]])
+        >>> layers.reduce(1, 3)
```

Comment:
Suggest adding a comment pointing out that (if I'm right) this combines the top 2 layers, which are 1 and 2, because layers are numbered upward from bottom (so the deepest layer is 0).